app/helpers/cart_helper.rb

```diff
@@ -0,0 +1,30 @@
+module CartHelper
+  def get_name_product pro_id
+    product = Product.find_by(id: pro_id)
+    product&.name
+  end
+
+  def get_price_product pro_id
+    product = Product.find_by(id: pro_id)
+    product&.price
+  end
+
+  def get_category_product pro_id
+    product = Product.find_by id: pro_id
+    product&.category_name
+  end
+
+  def count_item
+    @count = @carts.sum(&:second)
+    @count
+  end
+
+  def total_sum
+    sum = 0
+    products = Product.where id: @carts.map(&:first)
+    @carts.each do |(pro_id, qty)|
+      sum += products.find{|pro| pro.id == pro_id.to_i}.price * qty.to_i
```

Comment:
ko dùng find em nhé, vì sẽ có thể raise exception

em viết scope, where IN ids nhé, lấy ra tất cả product với id là từ key của hash cart, sau đó select chỉ lấy field price rồi dùng inject là method hỗ trợ của collection ấy , ko hiểu thì hỏi chị

Reply:
dạ để em sửa lại, bữa anh thâm đến chỉ em nói chỗ này nó sẽ không thể nil được nên dùng find, để em sửa lại như chị

Reply:
Chị Vân ơi em làm tới đoạn inject thì gặp vấn đề là em phaỉ nhân price với qty nữa nên inject không dùng được ạ